app/components/documents/document-card.js

```diff
@@ -172,9 +171,19 @@ export default class DocumentsDocumentCardComponent extends Component {
 
   @task
   *loadSignatureRelatedData() {
-    this.signMarkingActivity = yield this.args.piece.belongsTo('signMarkingActivity').reload();
-    const signSubcase = yield this.signMarkingActivity?.signSubcase;
-    this.signFlow = yield signSubcase?.signFlow;
+    const decisionActivity = yield this.args.piece.decisionActivity;
+    if (decisionActivity) {
+      const signFlow = yield this.store.queryOne('sign-flow', {
+        'filter[decision-activity][:id:]' : decisionActivity.get('id'),
+      });
+      this.signFlow = signFlow;
+      const signSubcase = yield this.signFlow.signSubcase;
+      this.signMarkingActivity = yield signSubcase.signMarkingActivity;
+    } else {
+      this.signMarkingActivity = yield this.args.piece.belongsTo('signMarkingActivity').reload();
+      const signSubcase = yield this.signMarkingActivity?.signSubcase;
+      this.signFlow = yield signSubcase?.signFlow;
+    }
```

Comment:
I don't understand this if-else block. If the piece has a decision activity, why do we need to fetch the sign flow through it?

This will also give unexpected results when an agendaitem has multiple pieces (or a BF) that are marked to be signed (because this query will return a random sign flow, not necessarily the one related to the current piece.

Can't we just use the logic in the else branch for all pieces?